src/libs/decoders/SDL_sound.h

```diff
@@ -729,6 +729,18 @@ SNDDECLSPEC int SDLCALL Sound_Seek(Sound_Sample *sample, Uint32 ms);
 
 #ifdef __cplusplus
 }
+
+inline Sound_SampleFlags operator|(Sound_SampleFlags a, Sound_SampleFlags b)
+{return static_cast<Sound_SampleFlags>(static_cast<int>(a) | static_cast<int>(b));}
+
+inline Sound_SampleFlags& operator|= (Sound_SampleFlags& a, Sound_SampleFlags b)
+{ return (Sound_SampleFlags&)((int&)a |= static_cast<int>(b)); }
+
+inline Sound_SampleFlags operator& (Sound_SampleFlags a, Sound_SampleFlags b)
+{ return (Sound_SampleFlags)((int)a & (int)b); }
+
+inline Sound_SampleFlags& operator&= (Sound_SampleFlags& a, Sound_SampleFlags b)
+{ return (Sound_SampleFlags&)((int&)a &= (int)b); }
```

Comment:
Bleh, I don't like overloading operators in C++ for stuff like this…

But this is one of those areas where C and C++ are incompatible with each other; in C these operators *just work*, but modern C++ require `Sound_SampleFlags` to be `enum class : int` (which makes it work like C, without being C - hurray :/).

One way of working around this would be to replace type field `flags` in `Sound_Sample` to `int` or `unsigned int`.

If you don't want to change the type for `flags` field and these operators are here to stay, then fix the formatting, please :)

Reply:
Yes, this one was quite ugly (it would be nice to just move to straight C++ so we're not shoving nice ideas like enum's into fixed-length bytes to satisfy both the C and C++ world) -- your approach nicely cleans up the need for those operators, and given our flag range only covers 8-bits, we've got plenty of room to grow if we need another 24 flags.  

Fixed in fd5c90b7084cad87cd9b82b0df9d9904e17ee1e8